services/listen_dnsport.c

```diff
@@ -4690,6 +4695,10 @@ doq_ctx_server_setup(struct doq_server_socket* doq_socket)
 	SSL_CTX_set_quic_method(ctx, doq_socket->quic_method);
 #endif
 	return ctx;
+#else /* HAVE_NGHTTP2 */
+	(void)key; (void)pem; (void)verifypem;
+	return NULL;
+#endif /* HAVE_NGHTTP2 */
```

Comment:
If the check is for ngtcp2 that would have to be noted here too.
```suggestion
#endif /* HAVE_NGTCP2 */
```